src/lib/hooks/useMatomo/index.tsx

```diff
@@ -0,0 +1,53 @@
+import { useRouter } from "next/router";
+import { useEffect } from "react";
+
+const MATOMO_URL =
+  process.env.NEXT_PUBLIC_MATOMO_URL || "https://piwik.example.com";
+const MATOMO_SITE_ID = process.env.NEXT_PUBLIC_MATOMO_SITE_ID || "1";
+
+const createImageNoscript = (pathname = ""): HTMLElement => {
```

Comment:
I'm not entirely understanding how these lines make the tracking work (even though I know it works). Do you have a pointer where to find documentation about this? Or is it a completely custom implementation? If too complex to put in a sentence or two, we can also sync about this in person sometime.

Reply:
It basically creates a new image each time the page changes and adds it to the DOM. The image is then loaded and is counted as a view in matomo. This is a custom implementation adapted to next.js. I'd be glad to explain in greater detail once we see each other! :)

Reply:
Ahh, I see. Yes, I'd be curious to know more about it :)